File: 102.二叉树的层序遍历.java
import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;
import java.util.Queue;

import javax.swing.tree.TreeNode;

import apple.laf.JRSUIUtils.Tree;

/*
 * @lc app=leetcode.cn id=102 lang=java
 *
 * [102] 二叉树的层序遍历
 */

// @lc code=start
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        Queue<TreeNode> queue=new LinkedList<TreeNode>();
        List<List<Integer>> nums=new ArrayList<>();
        queue.offer(root);
        while(!queue.isEmpty()){
            int size=queue.size();
            List<Integer> list=new ArrayList<>();
            for(int i=0;i<size;i++){
                TreeNode node=queue.poll();
                list.add(node.val);
                if(node.left!=null){
                    queue.add(node.left);
                }
                if(node.right!=null){
                    queue.add(node.right);
                }
            }
            nums.add(list);
            //queue.add()
        }
        return nums;
    }
    /***
     * 递归的使用深度优先遍历方法（利用系统栈）
     * @param root
     * @return
     */
    public List<TreeNode> DFS(TreeNode root){
        List<TreeNode> list=new ArrayList<>();
        if(root==null){
            return null;
        }
        list.addAll(DFS(root.left));
        list.addAll(DFS(root.right));
    }

}
// @lc code=end

File: 102.二叉树的层序遍历.py
'''
Author: your name
Date: 2021-04-20 21:04:55
LastEditTime: 2021-04-20 21:38:02
LastEditors: Please set LastEditors
Description: In User Settings Edit
FilePath: /leetcode/102.二叉树的层序遍历.py
'''
#
# @lc app=leetcode.cn id=102 lang=python3
#
# [102] 二叉树的层序遍历
#

# @lc code=start
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        if not root:
            return []
        tree_queue=[]
        nums=[]
        tree_queue.insert(0, root)
        while len(tree_queue)!=0:
            size=len(tree_queue)
            temp=[]
            for i in range(size):
                current_node=tree_queue.pop()
                temp.append(current_node.val)
                if current_node.left:
                    tree_queue.insert(0, current_node.left)
                if current_node.right:
                    tree_queue.insert(0, current_node.right)
            nums.append(temp)
        return nums
        
# @lc code=end

